docs/src/appendix/chisel-plugin.md

```diff
@@ -0,0 +1,76 @@
+# Notes on the Compiler Plug-in
+
+The Chisel plugin provides some operations that are too difficult, or not possbile, 
+to implement through regular Scala code.
+
+## Compiler plugin operations
+1. Automatically generates the `cloneType` methods of Bundles
+2. Records that this plugin ran on a bundle by adding a method `_usingPlugin: Boolean = true`
+3. Generates a function that returns list of the hardware fields in a bundle `_elementsImpl`
+
```

Comment:
4. Future work -- detecting Seq[Data] in Bundles? I'm confused about what that section is saying

Reply:
This section was just enumerating the things the plugin currently does.
I've added a bit more verbiage to this effect.